usagov/usagov-2021
# Usagov 1643 implement lazy loading

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1643

## Description

Added lazysizes library and Lazy Load Druple module
Added requirements and scripts to composer.json 
Added 'loading="lazy"' attribute to the<img> tags for social media inline images on USAgov homepage
Added "lazy-loading" attribute to to background CSS images for the icons on the USAgov homepage on the twig template
Added a lazyloadBgImage.js script to lazy-load the background CSS images for the icons on the USAgov homepage
Removed unnecessary background image properties from the CSS for the USAgov homepage icons

## Type of Changes

- [ x] New Feature
  - Add screenshot showing what it should look like
- [x ] Drupal Config (requires "drush cim")
- [ x] New Modules (requires rebuild)

## Testing Instructions


Run bin.drush cim
Run docker compose up
Go to localhost > inspect > Network > click "img" > reload webpage > scroll down through page slowly and watch images get downloaded as you scroll

Also can go localhost > inspect  > Lighthouse > run page analyze > double check that "defer offscreen images" does not appear in the problematic audits > check that "defer offscreen images" is included in the "passed audits" drop down

### Requires New Config
- [ x] Yes

### Requires New Content
- [x ] Yes

### Validation Steps
- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review


- [ x] Adds/updates software (including a library or Drupal module)


## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
